CI build android on ubuntu-latest failed

# Issue description
## Error logs:
ld.lld: error: non-exported symbol '__aeabi_ldivmod' in '/usr/local/lib/android/sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/lib/clang/18/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_ldivmod.S.o)' is referenced by DSO 'src/.libs/libzmq.so'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:4155: perf/local_lat] Error 1
make[1]: Leaving directory '/home/runner/work/libzmq/libzmq/libzmq'
make: *** [Makefile:8442: install-recursive] Error 1
Error: Process completed with exit code 1.

More details: https://github.com/zeromq/libzmq/actions/runs/10338824667/job/28617425863

## Problem analysis:
The current project only supports android-ndk-r25, while ubuntu-latest provides android-ndk-r27 by default.

## Solution:
Fix the NDK version in the CI script to android-ndk-r25 before the project adapts to other NDK versions.

--- a/.github/workflows/CI.yaml
+++ b/.github/workflows/CI.yaml
@@ -34,6 +34,7 @@ jobs:
             POLLER: poll
           - os: ubuntu-latest
             BUILD_TYPE: android
+            NDK_VERSION: android-ndk-r25
             DRAFT: disabled
           - os: ubuntu-latest
             BUILD_TYPE: coverage
@@ -148,6 +149,8 @@ jobs:
       USE_NSS: ${{ matrix.USE_NSS }}
       VMCI: ${{ matrix.VMCI }}
       POLLER: ${{ matrix.POLLER }}
+      NDK_VERSION: ${{ matrix.NDK_VERSION }}
+      ANDROID_NDK_ROOT: /tmp/${{ matrix.NDK_VERSION }}
     steps:    
     - name: Add msbuild to PATH
       uses: microsoft/setup-msbuild@v1.0.2

